include/caffe/layers/softmax_loss_layer.hpp

```diff
@@ -119,6 +119,10 @@ class SoftmaxWithLossLayer : public LossLayer<Dtype> {
   bool has_ignore_label_;
   /// The label indicating that an instance should be ignored.
   int ignore_label_;
+  /// Whether to only care about an instance with certain label.
+  bool has_only_care_label;
+  /// The only label we care about in this loss-layer.
+  int only_care_label;
```

Comment:
Class member variables should end with an underscore. Side note: why doesn't `lint` detect this?

Also, if I may suggest, a better name for this field would be `select_label`, or `specify_label` - to make it extra clear that this means: _select just these instances for the loss calculation_. Or maybe someone has a better idea?

Reply:
thank you for you side note.

Reply:
Thank you for you advice. I will make a new request named "Enhance SoftmaxWithLoss by adding select_label. #6324".
Please review the new one.